src/Tasks/Microsoft.NET.Build.Tasks/GenerateDepsFile.cs

```diff
@@ -103,6 +113,11 @@ protected override void ExecuteCore()
                 .WithReferenceAssembliesPath(FrameworkReferenceResolver.GetDefaultReferenceAssembliesPath())
                 .Build();
 
+            if (compileFilesToSkip.Any() || runtimeFilesToSkip.Any())
+            {
+                dependencyContext = TrimFilesToSkip(dependencyContext);
```

Comment:
Once you have some more tests to reduce the risk you should do this.  I suspect the current implementation is very inefficient since it reconstructs the entire dependency context object graph.  File an issue?